### DISCUSSION TITLE
Add B2B model

### DISCUSSION THREAD

**Discussion by Fastmancz** [Paymenter/Paymenter#484] [Ideas]
Currently Paymenter only supports payment for non-business individuals (B2C), but what if the customer is a business entity (B2B)?

If a non-business customer buys my services (B2C), I pay tax for them in Europe (OSS - One Stop Shop). The tax is calculated according to the customer's residence. If it is a third country (USA, Canada, Australia - outside Europe), the tax is not included.

If the purchase is made by a company (B2B) and it is from Europe, the tax is paid by the company - if it has a VAT ID filled in (e.g. FR123456789), it will appear on the invoice (VAT ID: 0% - Reverse charge). If the company is from a third country, the tax is not included.

I don't know if this applies everywhere across Europe. What I am writing here applies to the Czech Republic.

![vatid_b2b](https://github.com/user-attachments/assets/81d8c938-bca1-4faa-adc5-8f7df7f0e2dc)

**Comment by bitUpsurge** [comment]
A must have
